analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/Extensions/IOperationExtensions.cs

```diff
@@ -38,6 +38,15 @@ internal static class IOperationExtensions
                 _ => null
             };
 
+        internal static ISymbol ResolveTrackedSymbol(this IOperation operation, ProgramState state) =>
+            operation?.Kind switch
+            {
+                OperationKindEx.FlowCaptureReference => state.ResolveCapture(operation) is { } capture && capture != operation ? capture.ResolveTrackedSymbol(state) : null,
```

Comment:
`ResolveCapture` returns the FlowReferenceOperation if the reference can not be resolved, which would lead to an endless loop here.